yarn-project/pxe/src/pxe_service/pxe_service.ts

```diff
@@ -859,28 +859,42 @@ export class PXEService implements PXE {
     eventMetadata: EventMetadata<T>,
     from: number,
     limit: number,
-    ivpk: Point = Point.ZERO,
+    vpks: Point[] = [],
   ): Promise<T[]> {
     if (type.includes(EventType.Encrypted)) {
-      return this.getEncryptedEvents(from, limit, eventMetadata, ivpk);
+      return this.getEncryptedEvents(from, limit, eventMetadata, vpks);
     }
 
     return this.getUnencryptedEvents(from, limit, eventMetadata);
   }
 
-  async getEncryptedEvents<T>(from: number, limit: number, eventMetadata: EventMetadata<T>, ivpk: Point): Promise<T[]> {
+  async getEncryptedEvents<T>(
```

Comment:
Think it would be useful to throw here if `vpks` is empty since we would not be able to decrypt anything, so it is essentially a no-op, where we might as well give a neat error instead of having him wait to fetch the blocks from node and then don't find anything 👀